computation-slurm/src/main/java/com/powsybl/computation/slurm/TaskStore.java

```diff
@@ -23,293 +20,105 @@
 
     private static final Logger LOGGER = LoggerFactory.getLogger(TaskStore.class);
 
-    // workingDir<--->Task of computation
-    private Map<String, TaskCounter> workingDirTaskMap = new HashMap<>();
-    private Map<String, Long> workingDirFirstJobMap = new HashMap<>();
-    private Map<CompletableFuture, String> futureWorkingDirMap = new HashMap<>();
-    private Map<String, CompletableFuture> workingDirFutureMap = new HashMap<>();
-    private ReadWriteLock taskLock = new ReentrantReadWriteLock();
+    private Map<String, SlurmTask> taskByDir = new ConcurrentHashMap<>();
+    private Map<CompletableFuture, SlurmTask> taskByFuture = new ConcurrentHashMap<>();
 
-    private Map<Long, Long> jobDependencies = new HashMap<>();
-    private ReadWriteLock jobDependencyLock = new ReentrantReadWriteLock();
+    private final StoreCleaner cleaner;
 
-    private Map<Long, List<Long>> batchIds = new HashMap<>();
-    private ReadWriteLock batchIdsLock = new ReentrantReadWriteLock();
-
-    private Set<Long> tracingIds = Collections.newSetFromMap(new ConcurrentHashMap<>());
-
-    TaskCounter getTaskCounter(String workingDir) {
-        taskLock.readLock().lock();
-        try {
-            return workingDirTaskMap.get(workingDir);
-        } finally {
-            taskLock.readLock().unlock();
-        }
+    TaskStore(int cleanTime) {
+        cleaner = new StoreCleaner(cleanTime);
+        cleaner.start();
     }
 
-    TaskCounter getTaskCounter(CompletableFuture future) {
-        taskLock.readLock().lock();
-        try {
-            String workingDir = futureWorkingDirMap.get(future);
-            return workingDirTaskMap.get(workingDir);
-        } finally {
-            taskLock.readLock().unlock();
-        }
+    void add(SlurmTask task) {
+        taskByDir.put(task.getId(), task);
+        taskByFuture.put(task.getCompletableFuture(), task);
     }
 
-    Long getFirstJobId(CompletableFuture future) {
-        taskLock.readLock().lock();
-        try {
-            String workingDir = futureWorkingDirMap.get(future);
-            return workingDirFirstJobMap.get(workingDir);
-        } finally {
-            taskLock.readLock().unlock();
-        }
+    public Map<String, SlurmTask> getTaskByDir() {
```

Comment:
I think this should be at least package private (ideally private, but it's used for tests).